app/src/main/kotlin/com/wire/android/notification/broadcastreceivers/DynamicReceiversManager.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Wire
+ * Copyright (C) 2025 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.android.notification.broadcastreceivers
+
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import com.wire.android.appLogger
+import com.wire.android.emm.ManagedConfigurationsReceiver
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Manages dynamic registration and unregistration of broadcast receivers.
+ * This are receivers that are active while the app is in foreground only.
+ */
+@Singleton
+class DynamicReceiversManager @Inject constructor(
```

Comment:
Added this manager/helper class to reduce the code we are adding in `WireActivity` it is already ~700 lines and can be reused later for other broadcast receivers we might use.